### PR TITLE
java: slow tests

### PR DESCRIPTION
**Pull request by hdonnay**
I got really annoyed with the speed of the `java` tests, so I worked out how to not use the entire `osv` package and instead use fixtures.

### CODE REVIEW

**Comment by hdonnay** [comment]
ok  	github.com/quay/claircore/java	30.435s	coverage: 28.2% of statements

This normally takes ~500s

File: java/matcher_test.go [hunk 1]
```diff
@@ -5,16 +5,19 @@ import (
...
 )
 
 func TestVulnerable(t *testing.T) {
 	matcher := &Matcher{}
 
 	testcases := []struct {
-		name   string
 		record *claircore.IndexRecord
 		vuln   *claircore.Vulnerability
+		name   string
```
Comment (RTann) [inline java/matcher_test.go:20]: just curious why move this down? Seems like it tends to be the first thing set

File: java/matcher_test.go [hunk 2]
```diff
@@ -199,3 +202,7 @@ func TestVulnerable(t *testing.T) {
 		})
 	}
 }
+
+func TestMatcher(t *testing.T) {
+	test.RunMatcherTests(zlog.Test(context.Background(), t), t, "testdata/matcher", new(Matcher))
```
Comment (RTann) [inline java/matcher_test.go:207]: so when this runs, it keeps the current directory context? (as in, it finds `testdata/matcher` relative to the `java` directory and not the `test` directory. I didn't realize that, cool

File: java/testdata/matcher/Jackson-Databind.txtar
Comment (RTann) [inline java/testdata/matcher/Jackson-Databind.txtar:1]: just curious do you use a tool to create this or is this a decent amount of manual copy/paste + adding headers yourself?

**Review by RTann — APPROVED** [review]
curious are there plans to use this for other languages, too?

**Comment by hdonnay** [comment]
> curious are there plans to use this for other languages, too?

Yeah, I'm about to slap this on the other tests that currently use the `osv` updater.

File: java/matcher_test.go (hunk 1 above)
Comment (hdonnay) [reply java/matcher_test.go:20]: My editor complained about the layout causing extra padding, so I just let it rearrange the struct. The structs are all keyed, so it doesn't matter to the declarations.

File: java/testdata/matcher/Jackson-Databind.txtar
Comment (hdonnay) [reply java/testdata/matcher/Jackson-Databind.txtar:1]: This was taking the output VulnerabilityReport and chopping it up, and taking the old fixture. We could probably write a tool to take a vulnerability report and spit out something close to this final, pretty output.

**Pull request merged**